library/ssl_msg.c

```diff
@@ -1549,11 +1575,23 @@ int mbedtls_ssl_decrypt_buf( mbedtls_ssl_context const *ssl,
         return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
     }
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+    if( transform->minor_ver == MBEDTLS_SSL_MINOR_VERSION_4 )
+    {
+        /* Remove inner padding and infer true content type. */
+        ret = ssl_parse_inner_plaintext( data, &rec->data_len,
```

Comment:
Suggestion: And again, it it simpler to have one call to `ssl_parse_inner_plaintext()` with a more complex condition?

Reply:
I tried and found it a bit difficult to read, so preferred the present solution.